packages/@expo/cli/src/start/startAsync.ts

```diff
@@ -70,7 +72,20 @@ export async function startAsync(
 
   setNodeEnv(options.dev ? 'development' : 'production');
   require('@expo/env').load(projectRoot);
-  const { exp, pkg } = profile(getConfig)(projectRoot);
+  const { exp, pkg, hasUnusedStaticConfig, dynamicConfigPath, staticConfigPath } =
+    profile(getConfig)(projectRoot);
+
+  if (hasUnusedStaticConfig) {
```

Comment:
Move this warning into getConfig behind an option so it stays unified.

Reply:
We don't currently output anything to console in `getConfig`. Is the main goal here consistency in the warning message if we choose to display it other places where we process the config? Wondering if an alternative would be to return another key with warnings, so consumers could use or reformat those as they see fit.

However, if this is too prescriptive to warn in this context, I should avoid the refactor and just delete this part entirely. I could reserve any actual warnings for Doctor, where we're more prescriptive, but also less in-your-face ("this __might__ be an issue with your project").